### ISSUE TITLE
Lint/UnreachableLoop false positive when array for each loop represented as clause

### ISSUE THREAD

**Issue by Valentyna** [rubocop/rubocop#9804]
See 4th line
```
  # compare parsed json (nested hash-array combination)
  # @return [Hash/Array] difference if there is one
  def difference_between_hashes(a, b) 
    (a.keys | b.keys).each do |k|  # Lint/UnreachableLoop: This loop will have at most one iteration.
      if a[k] != b[k]
        diff = if a[k].is_a?(Hash) && b[k].is_a?(Hash)
                 difference_between_hashes(a[k], b[k])
               elsif a[k].is_a?(Array) && b[k].is_a?(Array)
                 (a[k] - b[k]) + (b[k] - a[k])
               else
                 [a[k], b[k]]
               end
      end
      return diff
    end
  end
```

## Expected behavior

When running method - loop in fact iterates over entire array of keys. No offense of such kind should be reported

## RuboCop version

```
$ [bundle exec] rubocop -V
1.14.0 (using Parser 3.0.1.1, rubocop-ast 1.5.0, running on ruby 3.0.0 x86_64-darwin20)
```

**Comment by tejasbubane** [comment]
@Valentyna This is not a false positive. That loop will run only once because of the `return`.

```
irb(main):005:1* def foo
irb(main):006:2*   [1,2,3,4].each do |n|
irb(main):007:2*     puts n
irb(main):008:2*     return 12
irb(main):009:1*   end
irb(main):010:0> end
=> :foo
irb(main):011:0> foo
1
=> 12
```

Remove `return` and the offense should go off.
